Sources/Build/BuildOperation.swift

```diff
@@ -237,6 +237,9 @@ public final class BuildOperation: PackageStructureDelegate, SPMBuildCore.BuildS
 
     /// Perform a build using the given build description and subset.
     public func build(subset: BuildSubset) throws {
+        if buildParameters.shouldSkipBuilding {
+            return
+        }
```

Comment:
nit: I'd expect this to use `guard` for early returns
```suggestion
        guard !buildParameters.shouldSkipBuilding else {
            return
        }
```